api/renewer.go

```diff
@@ -0,0 +1,280 @@
+package api
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func init() {
+	// Seed the random generator
+	rand.Seed(time.Now().Unix())
```

Comment:
You could store a Rand object in the `Renewer` struct so that you can independently seed each `Renewer` and avoid the side effect on import. https://golang.org/pkg/math/rand/#Rand